Src/ILGPU/Backends/OpenCL/CLCodeGenerator.Values.cs

```diff
@@ -296,7 +296,7 @@ public void Visit(AtomicCAS atomicCAS)
                 statement.BeginArguments();
                 statement.AppendAtomicCast(atomicCAS.ArithmeticBasicValueType);
                 statement.AppendArgument(target);
-                statement.AppendArgumentAddress(value);
+                statement.AppendArgumentAddressWithCast(value, "__generic " + CLTypeGenerator.GetBasicValueType(atomicCAS.ArithmeticBasicValueType) + " *");
```

Comment:
nit-picking: I think it would be a bit nicer to shorten this line, as it has significantly more than 100 characters and is not a comment line. Also, it would be great if we could use the `CLInstructions.DereferenceOperation` instead of the hard-coded `*`. How about creating a more generic function that generates this cast expression within the `CLInstructions` class?